interfaces/cython/cantera/composite.py

```diff
@@ -516,32 +516,38 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
         reserved = self.__dir__()
 
         self._extra = {}
-        if isinstance(extra, dict):
-            for name, v in extra.items():
+
+        if isinstance(extra, str):
+            extra = [extra]
+
+        if isinstance(extra, (list, tuple, np.ndarray)) and all(isinstance(name, str) for name in extra):
+            #if extra is a list and all elements in extra are of type str
```

Comment:
Can you delete this comment? Well, given the changes I suggested above it won't be relevant anymore. But even if that check were staying, this comment is explaining **what** rather than **why**. People who read the code on the previous line will already be able to tell that this branch will be executed if extra is a list-like and all elements are of type str. But notice that this comment has already become out-of-date, since `extra` could be more than just a `list` based on the check. You can read more about commenting for why instead of what in many places online, e.g., https://dev.to/andreasklinger/comments-explain-why-not-what-and-2-more-rules-on-writing-good-comments Just leaving this comment for learning purposes :smiley:

Reply:
This was quite resourceful thank you!! 😄😄.

The entire block could be deleted with the new suggestions